src/schema.rs

```diff
@@ -243,12 +289,12 @@ impl Schema {
             lookup.insert(field.name.clone(), field.position);
         }
 
-        Ok(Schema::Record(Rc::new(RecordSchema {
+        Ok(Schema::Record {
             name,
             doc: complex.doc(),
             fields,
             lookup,
-        })))
+        })
     }
 
     fn parse_enum(complex: &Map<String, Value>) -> Result<Self, Error> {
```

Comment:
It seems you forgot documentation for `parse_enum`. Even if it's private, you put docstrings everywhere, so I assume this slipped through.